### PR TITLE
[imaging_browser] Convert to "real" LORIS module

### PR DESCRIPTION
**Pull request by driusan**
This converts the imaging browser to a proper LORIS module. It adds a
namespace and Module class descriptor.

### CODE REVIEW

File: php/libraries/NDB_Caller.class.inc [hunk 1]
```diff
@@ -128,6 +128,22 @@ class NDB_Caller
...
+            // in React.
+            if (method_exists($this->page, 'getControlPanel')) {
+            $this->controlPanel = $this->page->getControlPanel();
+            }
+            if (method_exists($this->page, 'getFeedbackPanel')) {
+                $this->feedbackPanel = $this->page->getFeedbackPanel(
+                    $_REQUEST['candID'],
+                    $_REQUEST['sessionID'] ?? null
+                );
+            }
```
Comment (driusan) [inline php/libraries/NDB_Caller.class.inc]: This is the same as/copy/pasted from #3142, one of them will have conflicts and need to be rebased, but it's necessary for the control panel to work..

**Comment by xlecours** [comment]
@driusan modules/imaging_browser/php/Imaging_Session_ControlPanel.class.inc should be lower case

**Comment by driusan** [comment]
@xlecours It's done.

File: modules/imaging_browser/help/imaging_browser.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+New (have not had any QC done), Pending (QC is not finalized) or both.
+
+After specifying search criteria in the selection filters, click
+"Show Data" to display results in the data table below, with one
+row per relevant scanning session. Click "Clear Form" to reset all
+selection filters.
+
+Sort the data table results by clicking on any column header.  The
+date of first acquisition of all scans for the timepoint is indicated
+under the "First Acq Date" column.  The "Last QC" column shows the
```
Comment (MounaSafiHarab) [inline modules/imaging_browser/help/imaging_browser.md]: The column is now called 'First Acquisition"

File: modules/imaging_browser/help/imaging_browser.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+date of the most recent Quality Control (QC) on any scan in the
+session. New scans awaiting Quality Control (QC) will be marked
+"NEW" (in red) under the "New Data" column.
+
+To download, visualize, add QC annotation or view details of imaging
+files for a candidate-timepoint, select a dataset type the "Links"
```
Comment (MounaSafiHarab) [inline modules/imaging_browser/help/imaging_browser.md]: something is missing here; 'select a dataset type UNDER the "LInks" column' perhaps?

File: modules/imaging_browser/help/imaging_browser.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+row per relevant scanning session. Click "Clear Form" to reset all
+selection filters.
+
+Sort the data table results by clicking on any column header.  The
+date of first acquisition of all scans for the timepoint is indicated
+under the "First Acq Date" column.  The "Last QC" column shows the
+date of the most recent Quality Control (QC) on any scan in the
+session. New scans awaiting Quality Control (QC) will be marked
+"NEW" (in red) under the "New Data" column.
+
```
Comment (MounaSafiHarab) [inline modules/imaging_browser/help/imaging_browser.md]: @llevitis @driusan 

Here we can also add a sentence about the fact that the modalities for which we want the QC Status displayed in separate columns of the table can be configured in the Admin/Config/Imaging Modules on a project level. And that the "NEW" flag reflects a "NEW" in the New Data column reflects an acquisition in any of those chosen modalities.

File: modules/imaging_browser/help/viewsession.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+Buttons at the bottom of each scan panel allow users to view/edit
+QC Comments, download the scan as a Minc file, or viewing the scans
+from a Longitudinal View which will launch the BrainBrowser. Click
+the QC Comments button to open up a pop-up window in which users
+with appropriate permissions may select QC issues and enter comments
+regarding intensity, movement artifacts, coverage and overall
+feedback on the selected scan.
+
+In the left sidebar
```
Comment (MounaSafiHarab) [inline modules/imaging_browser/help/viewsession.md]: @driusan 

just a small formatting issue here; should this "In the left sidebar" be a main heading/title?

**Comment by MounaSafiHarab** [comment]
@driusan 

help text in viewsession is not showing up for some reason...

viewsession help is now showing up for some reason

**Comment by xlecours** [comment]
@driusan regarding Mouna's last comment, `viewsession.md` should be `viewSession.md`

File: modules/imaging_browser/help/imaging_browser.md (hunk 4 above)
Comment (driusan) [reply modules/imaging_browser/help/imaging_browser.md]: Is that something that we want to be displaying to all users in the frontend?

**Comment by driusan** [comment]
@MounaSafiHarab I renamed the file as @xlecours suggested, does that help?

File: modules/imaging_browser/help/imaging_browser.md (hunk 2 above)
Comment (driusan) [reply modules/imaging_browser/help/imaging_browser.md]: done

File: modules/imaging_browser/help/imaging_browser.md (hunk 3 above)
Comment (driusan) [reply modules/imaging_browser/help/imaging_browser.md]: done

File: modules/imaging_browser/help/viewsession.md (hunk 5 above)
Comment (driusan) [reply modules/imaging_browser/help/viewsession.md]: I had just copy/pasted it from the other PR, but I agree.. I most it to be part of the title.

**Comment by MounaSafiHarab** [comment]
@driusan 

now viewsession.md loads. It is good to know as far as I am concerned. Two tiny/minor comments in the viewsession help:

1) 'Navigation in left side bar' is weirdly formatted again!
imagbrowser1

2) when you talk about caveat/selected/QC; should we say there is also the SNR (Isomething like: the signal to noise ratio which is computed on structural modalities during the imaging insertion pipeline).

**Comment by driusan** [comment]
@MounaSafiHarab I fixed the formatting. I don't want to diverge to much from what's in @llevitis's SQL PR, so maybe we can merge this (with the format fixed) and then you can send a PR to propose changes about SNR?

**Comment by MounaSafiHarab** [comment]
@driusan 

formatting is good now!

**Pull request merged**